### PR TITLE
refactor: remove conditional autocomplete option return types

### PR DESCRIPTION
**Pull request by suneettipirneni**
**Please describe the changes this PR makes and why it should be merged:**

Remove conditional return types from autocomplete options. These did not work as intended.

**Status and versioning classification:**
- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating

### CODE REVIEW

File: packages/builders/src/interactions/slashCommands/mixins/ApplicationCommandOptionWithChoicesAndAutocompleteMixin.ts [hunk 1]
```diff
@@ -62,11 +62,7 @@ export class ApplicationCommandOptionWithChoicesAndAutocompleteMixin<T extends s
 		return this;
 	}
 
-	public setChoices<Input extends [name: string, value: T][]>(
-		choices: Input,
-	): Input extends []
-		? this & Pick<ApplicationCommandOptionWithChoicesAndAutocompleteMixin<T>, 'setAutocomplete'>
-		: Omit<this, 'setAutocomplete'> {
+	public setChoices(choices: [name: string, value: T][]) {
```
Comment (almeidx) [inline packages/builders/src/interactions/slashCommands/mixins/ApplicationCommandOptionWithChoicesAndAutocompleteMixin.ts:65]:
```suggestion
	public setChoices(choices: [name: string, value: T][]): this {
```

File: packages/builders/src/interactions/slashCommands/mixins/ApplicationCommandOptionWithChoicesAndAutocompleteMixin.ts [hunk 2]
```diff
@@ -83,11 +79,7 @@ export class ApplicationCommandOptionWithChoicesAndAutocompleteMixin<T extends s
 	 * Marks the option as autocompletable
 	 * @param autocomplete If this option should be autocompletable
 	 */
-	public setAutocomplete<U extends boolean>(
-		autocomplete: U,
-	): U extends true
-		? Omit<this, 'addChoice' | 'addChoices'>
-		: this & Pick<ApplicationCommandOptionWithChoicesAndAutocompleteMixin<T>, 'addChoice' | 'addChoices'> {
+	public setAutocomplete(autocomplete: boolean) {
```
Comment (almeidx) [inline packages/builders/src/interactions/slashCommands/mixins/ApplicationCommandOptionWithChoicesAndAutocompleteMixin.ts:82]:
```suggestion
	public setAutocomplete(autocomplete: boolean): this {
```

**Review by almeidx — COMMENTED** [review]
Consistency. Alternatively, the annotation should be removed from all the other ones

**Review by vladfrangu — APPROVED** [review]
Once you apply @almeidx's request, LGTM

**Review by kyranet — APPROVED** [review]

**Review by iCrawl — APPROVED** [review]

**Pull request merged**